src/radical/pilot/agent/scheduler/continuous_fifo.py

```diff
@@ -0,0 +1,51 @@
+
+__copyright__ = "Copyright 2013-2016, http://radical.rutgers.edu"
+__license__   = "MIT"
+
+
+from .continuous import Continuous
+
+# ------------------------------------------------------------------------------
+#
+# This is a simle extension of the Continuous scheduler which makes RP behave
+# like a FiFo: it will only really attempt to schedule units if all older units
+# (units with lower unit ID) have been scheduled.  As such, it relies on unit
+# IDs to be of the format `unit.%d'`.
+#
+class ContinuousFifo(Continuous):
+
+    # --------------------------------------------------------------------------
+    #
+    def __init__(self, cfg, session):
+
+        self.slots  = None
+        self._index = 0
```

Comment:
Functionally, starting with 0 doesn't look wrong. I think the default value for ```self._index``` should be -1 or None (basically not 0). I think it is easier to understand because when we start from 0, self._index remains 0 even after unit.000000 has been allocated. Would you agree?

Reply:
I think this became outdated but still applies for ```self._last = 0```.

Reply:
Ah, an off-by-one error!  You are correct :-)